File: src/state.ts
import { NewsItem, Script } from "./models.js";

export interface DirectorData {
    angle: string;
    title_hook: string;
    search_query: string;
    key_questions: string[];
}

export interface Metadata {
    title: string;
    thumbnail_title: string;
    description: string;
    tags: string[];
}

export interface PublishResults {
    youtube?: { status: string; video_id?: string };
    twitter?: { status: string; tweet_id?: string };
}

export interface AgentState {
    run_id: string;
    bucket: string;
    limit?: number;
    news?: NewsItem[];
    script?: Script;
    audio_paths?: string[];
    video_path?: string;
    thumbnail_path?: string;
    status?: string;
    director_data?: DirectorData;
    metadata?: Metadata;
    publish_results?: PublishResults;
    memory_context?: string;
}

File: src/state.py
from typing import List, TypedDict

from src.models import NewsItem, Script


class AgentState(TypedDict):
    run_id: str
    bucket: str
    limit: int
    news: List[NewsItem]
    script: Script
    audio_paths: List[str]
    video_path: str
    status: str
    trend_mode: bool
    trend_data: dict
    director_data: dict
    knowledge_context: dict
